Dockerfile.build

```diff
@@ -1,21 +1,14 @@
 FROM heroku/heroku:18-build
 
-# Google Container Registry Authentication for Knative Support
-ARG docker_cred_helper_url=https://github.com/GoogleCloudPlatform/docker-credential-gcr/releases/download/v1.5.0/docker-credential-gcr_linux_amd64-1.5.0.tar.gz
-RUN curl -sfL ${docker_cred_helper_url} | tar xz -C /usr/local/bin
-
 ARG pack_uid=1000
 ARG pack_gid=1000
-ENV CNB_USER_ID=${pack_uid}
-ENV CNB_GROUP_ID=${pack_gid}
 
-RUN mkdir -p /workspace
-RUN groupadd pack --gid ${pack_gid} && \
-    useradd -u ${pack_uid} -g ${pack_gid} -ms /bin/bash heroku
+RUN groupadd heroku --gid ${pack_gid} && \
+  useradd heroku -u ${pack_uid} -g ${pack_gid} -s /bin/bash -md /app
```

Comment:
I don't think `/app` is where I would make the home directory. I would probably leave it alone.

Reply:
That's where $HOME is today during build and inside a running dyno. Doing anything else seems weird.

Reply:
Nothing happens in the home directory in our current examples. The builder image uses `/home/heroku` but the run image uses `/workspace`. Now we would be using `/app` during build and `/workspace` in the run image right?

FWIW the CF ones use the default home directory of `/home/cnb` in both build and run images.

I wonder what the requirement for a writeable home directory is intended for. @hone?

> The image config's User field is set to a non-root user with a writable home directory.

Reply:
I think we could set $HOME to /home/heroku, but we'll still need `/app` to exist and be writeable by the heroku user, at least while we're maintaining support for shimmed buildpacks.